src/features/lightspeed/commands/providerCommands.ts

```diff
@@ -0,0 +1,292 @@
+              `✅ ${activeProvider.toUpperCase()} provider connected successfully!`,
+              {
+                detail: status.modelInfo
+                  ? `Model: ${status.modelInfo.name}`
+                  : undefined,
+              },
+            );
+          } else {
+            vscode.window.showErrorMessage(
+              `❌ ${activeProvider.toUpperCase()} provider connection failed: ${status.error}`,
+            );
+          }
+        },
+      );
+    } catch (error) {
+      vscode.window.showErrorMessage(
+        `Connection test failed: ${error instanceof Error ? error.message : "Unknown error"}`,
+      );
+    }
+  }
+
+  /**
+   * Configure LLM provider through guided setup
+   */
+  private async configureLlmProvider(): Promise<void> {
+    try {
+      const supportedProviders = providerFactory.getSupportedProviders();
+
+      // Step 1: Select provider type
+      const providerItems = supportedProviders.map((provider) => ({
+        label: provider.displayName,
+        description: provider.description,
+        detail: provider.type,
+        provider: provider,
+      }));
+
+      const selectedProvider = await vscode.window.showQuickPick(
+        providerItems,
+        {
+          placeHolder: "Select a LLM provider",
+          ignoreFocusOut: true,
+        },
+      );
+
+      if (!selectedProvider) {
+        return;
+      }
+
+      // Step 2: Configure provider settings
+      const lightspeedConfig =
+        vscode.workspace.getConfiguration("ansible.lightspeed");
+      await lightspeedConfig.update(
+        "provider",
+        selectedProvider.provider.type,
+        vscode.ConfigurationTarget.Workspace,
+      );
+
+      // Configure required fields using the main lightspeed config
+      for (const field of selectedProvider.provider.configSchema) {
+        if (field.required) {
+          const currentValue = lightspeedConfig.get(field.key, "");
+          const inputOptions: vscode.InputBoxOptions = {
+            prompt: field.label,
+            placeHolder: field.placeholder,
+            value: currentValue as string,
+            ignoreFocusOut: true,
+          };
+
+          if (field.type === "password") {
+            inputOptions.password = true;
+          }
+
+          const value = await vscode.window.showInputBox(inputOptions);
+
+          if (value === undefined) {
+            return; // User cancelled
+          }
+
+          await lightspeedConfig.update(
+            field.key,
+            value,
+            vscode.ConfigurationTarget.Workspace,
+          );
+        }
+      }
+
+      // Set default endpoint if available and not already set
+      if (
+        selectedProvider.provider.defaultEndpoint &&
+        !lightspeedConfig.get("apiEndpoint")
+      ) {
+        await lightspeedConfig.update(
+          "apiEndpoint",
+          selectedProvider.provider.defaultEndpoint,
+          vscode.ConfigurationTarget.Workspace,
```

Comment:
This updates the workspace config but what if the config it for folder or global settings. Should a check to inspect config type should be added here `const providerInspect = config.inspect<string>("provider");`